packages/database/lib/migrations/20241122212401_integration_backfill_missing_fields.cjs

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-var-requires */
+const path = require('node:path');
+const fs = require('node:fs');
+const yaml = require('js-yaml');
+
+exports.up = async function (knex) {
+    const providersPath = path.join(__dirname, '..', '..', '..', 'shared', 'providers.yaml');
+    const providers = yaml.load(fs.readFileSync(providersPath, 'utf8'));
```

Comment:
Is this reasonable? I had to be able to figure out the `auth_mode` and this was the most direct path I could come up with.

Reply:
Unfortunately we overload the usage of some of these fields in terms of types of auth so it's not the easiest to infer. See https://github.com/NangoHQ/nango/blob/9ddb80e8a9fc8e19dcf77f57d1e02042c4adf5f7/packages/webapp/src/pages/Integrations/providerConfigKey/Settings/components/App.tsx#L22-L23